cordyceps/src/list.rs

```diff
@@ -897,6 +913,17 @@ impl<T: Linked<Links<T>> + ?Sized> List<T> {
         }
     }
 
+    /// Returns an iterator over the items in this list, by pointer.
+    #[must_use]
+    pub fn raw_iter(&self) -> RawIter<'_, T> {
```

Comment:
hmm, should this mutably borrow `self`? since the entries can be mutated through the iterator..

it occurs to me that we _could_ even have a `RawIter` that immutably borrows the list and hands out `*const T`s, and a `RawIterMut` that mutably borrows the list and hands out `*mut T`s or `NonNull<T>`s. but, the `*const T` means we lose out on the `NonNull` guarantee and niche optimization...i am once again wishing that libcore had a `ConstNonNull` type thing...

Reply:
i think this _has_ to mutably borrow `self` since otherwise, it allows constructing a mutable reference from an immutable reference --- does miri not get mad if you do that? we might consider adding the `*const T` iterator as well.

Reply:
I think in my case it doesn't complain, likely due to some unsafe cell indirection, but it makes sense if the list owns all nodes it should require mut to allow full raw access.